src/fn_strings.cpp

```diff
@@ -65,20 +65,12 @@ namespace Sass {
     Signature quote_sig = "quote($string)";
     BUILT_IN(sass_quote)
     {
-      AST_Node_Obj arg = env["$string"];
-      // only set quote mark to true if already a string
-      if (String_Quoted* qstr = Cast<String_Quoted>(arg)) {
-        qstr->quote_mark('*');
-        return qstr;
-      }
-      // all other nodes must be converted to a string node
-      std::string str(quote(arg->to_string(ctx.c_options), '"'));
-      String_Quoted* result = SASS_MEMORY_NEW(String_Quoted, pstate, str);
+      const String_Constant* s = ARG("$string", String_Constant);
+      String_Quoted* result = SASS_MEMORY_NEW(String_Quoted, pstate, s->value(), '"');
```

Comment:
AFAICS you are setting the quotemark to `"` on creation, but it is reset one line below to `*`.
IMO you can just use the following and remove the lines below:
`return SASS_MEMORY_NEW(String_Quoted, pstate, s->value(), '*');`

Reply:
Do you know what the `*` quote mark means? Perhaps it should be `"`?

Reply:
Turns out the `String_Quoted` constructor does a bunch of non-trivial things and passing it a quote mark doesn't actually set it.

Updated, PTAL.

Reply:
A * should mean that on output we choose the best quotemarks automatically.